common/persistence/metered.go

```diff
@@ -67,6 +68,209 @@ func (r GetAllHistoryTreeBranchesResponse) Len() int {
 	return len(r.Branches)
 }
 
+// For responses that require metrics for payload size EstimatePayloadSizeInBytes() int should be defined.
+
+func (r *GetReplicationTasksResponse) EstimatePayloadSizeInBytes() int {
+	if r == nil {
+		return 0
+	}
+
+	size := int(unsafe.Sizeof(*r)) + len(r.NextPageToken)
+	for _, v := range r.Tasks {
+		if v == nil {
+			continue
+		}
+
+		size += int(unsafe.Sizeof(*v)) + len(v.DomainID) + len(v.WorkflowID) + len(v.RunID) + len(v.BranchToken) + len(v.NewRunBranchToken)
+	}
+
+	return size
+}
+
+func (r *GetTimerIndexTasksResponse) EstimatePayloadSizeInBytes() int {
+	if r == nil {
+		return 0
+	}
+
+	size := int(unsafe.Sizeof(*r)) + len(r.NextPageToken)
+	for _, v := range r.Timers {
+		if v == nil {
+			continue
+		}
+
+		size += int(unsafe.Sizeof(*v)) + len(v.DomainID) + len(v.WorkflowID) + len(v.RunID)
+	}
+
+	return size
+}
+
+func (r *GetTasksResponse) EstimatePayloadSizeInBytes() int {
+	if r == nil {
+		return 0
+	}
+
+	size := int(unsafe.Sizeof(*r))
+	for _, v := range r.Tasks {
+		if v == nil {
+			continue
+		}
+
+		size += int(unsafe.Sizeof(*v)) + len(v.DomainID) + len(v.WorkflowID) + len(v.RunID) + estimateStringMapSize(v.PartitionConfig)
+	}
+
+	return size
+}
+
+func (r *ListDomainsResponse) EstimatePayloadSizeInBytes() int {
+	if r == nil {
+		return 0
+	}
+
+	size := int(unsafe.Sizeof(*r)) + len(r.NextPageToken)
+
+	for _, v := range r.Domains {
+		if v == nil {
+			continue
+		}
+
+		size += int(unsafe.Sizeof(*v)) + estimateDomainInfoSize(v.Info) + estimateDomainConfigSize(v.Config) + estimateDomainReplicationConfigSize(v.ReplicationConfig)
+	}
+
+	return size
+}
+
+func estimateDomainInfoSize(info *DomainInfo) int {
+	if info == nil {
+		return 0
+	}
+
+	return int(unsafe.Sizeof(*info)) + len(info.ID) + len(info.Name) + len(info.Description) + len(info.OwnerEmail) + estimateStringMapSize(info.Data)
+}
+
+func estimateDomainConfigSize(info *DomainConfig) int {
+	if info == nil {
+		return 0
+	}
+
+	size := int(unsafe.Sizeof(*info)) + len(info.HistoryArchivalURI) + len(info.VisibilityArchivalURI)
+
+	asyncWorkflowConfigSize := int(unsafe.Sizeof(info.AsyncWorkflowConfig)) + len(info.AsyncWorkflowConfig.PredefinedQueueName) + len(info.AsyncWorkflowConfig.QueueType)
+	if info.AsyncWorkflowConfig.QueueConfig != nil {
+		size += len(info.AsyncWorkflowConfig.QueueConfig.Data)
+	}
+
+	binariesSize := 0
+	for key, value := range info.BadBinaries.Binaries {
```

Comment:
FYI: I don't think anything domain related will be very big, but the BadBinaries can be huge for the executions table (for concrete / Type 1 executions).